src/middlewares/http-logger.middleware.ts

```diff
@@ -0,0 +1,29 @@
+import { Inject, Injectable } from "@nestjs/common";
+import { Request, Response } from "express";
+import { Logger as WinstonLogger } from "winston";
+import { WINSTON_MODULE_NEST_PROVIDER } from "nest-winston";
+import { now } from "@wiredcraft/utils/comm.util";
+
+@Injectable()
+export class HttpLoggerMiddleware {
+  constructor(
+    @Inject(WINSTON_MODULE_NEST_PROVIDER) private readonly logger: WinstonLogger
+  ) {}
+
+  use(req: Request, res: Response, next: () => void) {
+    const { method, originalUrl: url } = req;
+
+    res.on("finish", () => {
+      const startTime = now("micro");
```

Comment:
can you explain why the `startTime` variable is placed inside the finish callback

Reply:
1. I meant to 'Aggregation' I think in current requirements user controller should has strong dependency with user service.
2. this a bug, I found this issue yesterday after I commit. it should be in upper block. and the reason why use micro time b/c of it is too fast when executing on local. if you seconds it will always be 0ms.